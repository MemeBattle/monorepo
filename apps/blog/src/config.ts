export const AMPLITUDE_API_KEY = process.env.NEXT_PUBLIC_AMPLITUDE_API_KEY
