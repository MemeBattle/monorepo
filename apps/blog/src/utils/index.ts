export { memeberToPostAuthor } from './memeberToPostAuthor'
export { formatDate } from './formatDate'
export { generateFullUrl } from './generateFullUrl'
