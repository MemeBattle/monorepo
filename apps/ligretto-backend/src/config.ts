import { readFileSync } from 'fs'
import { resolve } from 'path'

export const { SOCKET_PORT, PARTNER_ID, CAS_URL, LIGRETTO_CAS_KEY_PATH } = process.env

export const SOCKET_ROOM_LOBBY = 'SOCKET_ROOM_LOBBY'

const PATH_TO_KEY = LIGRETTO_CAS_KEY_PATH || resolve(__dirname, '../key.pem')

export const PUBLIC_KEY = readFileSync(PATH_TO_KEY).toString()
