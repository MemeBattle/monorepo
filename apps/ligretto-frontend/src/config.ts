export const WEBSOCKET_URL = process.env.REACT_APP_WEBSOCKET_HOST || 'localhost:3005'

export const MAX_CARDS_ON_TABLE = 12
export const TABLE_CARDS_PREFIX = 't'
