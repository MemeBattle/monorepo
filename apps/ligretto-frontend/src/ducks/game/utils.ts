import type { Player as SharedPlayer } from '@memebattle/ligretto-shared'

import type { User } from '../users/usersTypes'

export const mergePlayerAndUser = (player: SharedPlayer, user: User) =>
  ({
    status: player.status,
    isHost: player.isHost,
    username: user.isTemporary ? 'Guest' : user.username,
    id: user.casId,
    avatar: user.isTemporary ? '' : user.avatar,
    cards: player.cards,
    stackOpenDeck: player.stackOpenDeck,
  } as const)

export const STACK_OPEN_DECK_INDEX = 'STACK_OPEN_DECK_INDEX' as const
