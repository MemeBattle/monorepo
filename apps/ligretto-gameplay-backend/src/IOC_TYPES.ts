export const IOC_TYPES = {
  WebSocketHandler: Symbol.for('WebSocketHandler'),
  GameService: Symbol.for('GameService'),
  GameRepository: Symbol.for('GameRepository'),
  PlaygroundService: Symbol.for('PlaygroundService'),
  PlaygroundRepository: Symbol.for('PlaygroundRepository'),
  PlayerService: Symbol.for('PlayerService'),
  PlayerRepository: Symbol.for('PlayerRepository'),
  Gameplay: Symbol.for('Gameplay'),
  GameplayController: Symbol.for('GameplayController'),
  GameplayOutput: Symbol.for('GameplayOutput'),
  GamesController: Symbol.for('GamesController'),
  TechController: Symbol.for('TechController'),
  RoomRepository: Symbol.for('RoomRepository'),
  RoomService: Symbol.for('RoomService'),
  Emitter: Symbol.for('Emitter'),
  Database: Symbol.for('Database'),
  UserRepository: Symbol.for('UserRepository'),
  UserService: Symbol.for('UserService'),
  BotController: Symbol.for('BotController'),
  AuthService: Symbol.for('AuthService'),
  LigrettoCoreService: Symbol.for('LigrettoCoreService'),
}
