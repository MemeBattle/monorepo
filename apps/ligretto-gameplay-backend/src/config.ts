import { readFileSync } from 'node:fs'
import { resolve } from 'node:path'

import dotenv from '@codexsoft/dotenv-flow'

dotenv.config({ default_node_env: 'development', path: resolve(__dirname, '../../..'), silent: true })

export const { LIGRETTO_GAMEPLAY_SOCKET_PORT, CAS_PARTNER_ID = '', CAS_URL = '', LIGRETTO_GAMEPLAY_CAS_KEY_PATH, LIGRETTO_CORE_URL } = process.env

export const SOCKET_ROOM_LOBBY = 'SOCKET_ROOM_LOBBY'

const PATH_TO_KEY = LIGRETTO_GAMEPLAY_CAS_KEY_PATH || resolve(__dirname, '../key.pem')

export const PUBLIC_KEY = readFileSync(PATH_TO_KEY).toString()
