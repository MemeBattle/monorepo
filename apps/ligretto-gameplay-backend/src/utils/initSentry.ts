import * as Sentry from '@sentry/node'
import { nodeProfilingIntegration } from '@sentry/profiling-node'

Sentry.init({
  dsn: 'https://b9952c3fe0699ed45d67525f493b5b62@o1040475.ingest.us.sentry.io/4507005403725824',
  integrations: [nodeProfilingIntegration()],
  // Performance Monitoring
  tracesSampleRate: 1.0, //  Capture 100% of the transactions
  // Set sampling rate for profiling - this is relative to tracesSampleRate
  profilesSampleRate: 1.0,
  debug: true,
})

// setTimeout(() => {
//   Sentry.startSpan({ name: 'test' }, () => {
//     Sentry.startSpan({ name: 'tnested-test', op: 'mark.fn' }, () => {
//       console.log('test')
//     })
//   })
// }, 1000)
