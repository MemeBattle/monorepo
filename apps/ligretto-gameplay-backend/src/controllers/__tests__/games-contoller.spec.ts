import { createRoomEmitAction } from '@memebattle/ligretto-shared'

import type { GamesController } from '../games-controller'

import { IOC } from '../../inversify.config'
import { IOC_TYPES } from '../../IOC_TYPES'
import type { Database } from '../../database'
import { socketMockImpl } from '../../../test/utils'
import type { AnyAction } from '../../types/any-action'

describe('Games Controller', () => {
  it('should be defined', () => {
    const controller: GamesController = IOC.get(IOC_TYPES.GamesController)

    expect(controller).toBeDefined()
  })

  it('should create relevant state on create game', done => {
    const createGameService = jest.fn().mockReturnValue({ id: '1' })

    IOC.rebind(IOC_TYPES.LigrettoCoreService).toConstantValue({ createGameService })
    const controller: GamesController = IOC.get(IOC_TYPES.GamesController)
    const database: Database = IOC.get(IOC_TYPES.Database)

    const createGameAction = createRoomEmitAction({ name: 'createGame', config: {} }) as AnyAction

    const handleMessagePromise = controller.handleMessage(socketMockImpl, createGameAction) as unknown as Promise<void>

    handleMessagePromise.then(() => {
      database.get(db => {
        expect(db).toMatchSnapshot()

        done()
      })
    })
  })
})
