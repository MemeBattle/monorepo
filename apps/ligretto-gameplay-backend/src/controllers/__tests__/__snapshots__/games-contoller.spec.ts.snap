// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Games Controller createGame Should create relevant state on create game 1`] = `
Object {
  "games": Object {
    "1": Object {
      "config": Object {
        "dndEnabled": false,
        "playersMaxCount": 4,
        "startingDelayInSec": 4,
      },
      "id": "1",
      "name": "createGame",
      "players": Object {},
      "playground": Object {
        "decks": Array [],
        "droppedDecks": Array [],
      },
      "spectators": Object {},
      "status": "New",
    },
  },
  "users": Object {},
}
`;

exports[`Games Controller joinGame Should create relevant state on join room as first player 1`] = `
Object {
  "games": Object {
    "1": Object {
      "config": Object {
        "dndEnabled": false,
        "playersMaxCount": 4,
        "startingDelayInSec": 4,
      },
      "id": "1",
      "name": "createGame",
      "players": Object {
        "userId": Object {
          "cards": Array [],
          "id": "userId",
          "isHost": true,
          "ligrettoDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "stackDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "stackOpenDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "status": "DontReadyToPlay",
        },
      },
      "playground": Object {
        "decks": Array [],
        "droppedDecks": Array [],
      },
      "spectators": Object {},
      "status": "New",
    },
  },
  "users": Object {
    "userId": Object {
      "currentGameId": "1",
      "id": "userId",
      "socketIds": Array [
        "socketId",
      ],
    },
  },
}
`;

exports[`Games Controller joinGame Should create relevant state on join room by second connection 1`] = `
Object {
  "games": Object {
    "1": Object {
      "config": Object {
        "dndEnabled": false,
        "playersMaxCount": 4,
        "startingDelayInSec": 4,
      },
      "id": "1",
      "name": "createGame",
      "players": Object {
        "userId": Object {
          "cards": Array [],
          "id": "userId",
          "isHost": true,
          "ligrettoDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "stackDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "stackOpenDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "status": "DontReadyToPlay",
        },
      },
      "playground": Object {
        "decks": Array [],
        "droppedDecks": Array [],
      },
      "spectators": Object {},
      "status": "New",
    },
  },
  "users": Object {
    "userId": Object {
      "currentGameId": "1",
      "id": "userId",
      "socketIds": Array [
        "socketId",
        "socketId",
      ],
    },
  },
}
`;

exports[`Games Controller joinGame Should create relevant state on join room second player 1`] = `
Object {
  "games": Object {
    "1": Object {
      "config": Object {
        "dndEnabled": false,
        "playersMaxCount": 4,
        "startingDelayInSec": 4,
      },
      "id": "1",
      "name": "createGame",
      "players": Object {
        "secondUserId": Object {
          "cards": Array [],
          "id": "secondUserId",
          "isHost": false,
          "ligrettoDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "stackDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "stackOpenDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "status": "DontReadyToPlay",
        },
        "userId": Object {
          "cards": Array [],
          "id": "userId",
          "isHost": true,
          "ligrettoDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "stackDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "stackOpenDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "status": "DontReadyToPlay",
        },
      },
      "playground": Object {
        "decks": Array [],
        "droppedDecks": Array [],
      },
      "spectators": Object {},
      "status": "New",
    },
  },
  "users": Object {
    "secondUserId": Object {
      "currentGameId": "1",
      "id": "userId",
      "socketIds": Array [
        "secondUserSocketId",
      ],
    },
    "userId": Object {
      "currentGameId": "1",
      "id": "secondUserId",
      "socketIds": Array [
        "secondUserSocketId",
      ],
    },
  },
}
`;

exports[`Games Controller leaveFromRoomHandler Should create a relevant game state if one of two players leaved 1`] = `
Object {
  "games": Object {
    "1": Object {
      "config": Object {
        "dndEnabled": false,
        "playersMaxCount": 4,
        "startingDelayInSec": 4,
      },
      "id": "1",
      "name": "createGame",
      "players": Object {
        "userTwoId": Object {
          "cards": Array [],
          "id": "userTwoId",
          "isHost": true,
          "ligrettoDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "stackDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "stackOpenDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "status": "DontReadyToPlay",
        },
      },
      "playground": Object {
        "decks": Array [],
        "droppedDecks": Array [],
      },
      "spectators": Object {},
      "status": "Pause",
    },
  },
  "users": Object {
    "userOneId": Object {
      "currentGameId": "1",
      "id": "userOneId",
      "socketIds": Array [
        "socketId",
      ],
    },
    "userTwoId": Object {
      "currentGameId": "1",
      "id": "userTwoId",
      "socketIds": Array [
        "userTwoId",
      ],
    },
  },
}
`;

exports[`Games Controller leaveFromRoomHandler Should create a relevant state if last user disconnected 1`] = `
Object {
  "games": Object {},
  "users": Object {
    "userOneId": Object {
      "currentGameId": "1",
      "id": "userOneId",
      "socketIds": Array [
        "socketId",
      ],
    },
  },
}
`;

exports[`Games Controller leaveFromRoomHandler Should remove current socketId from user socket ids if user connected from few accounts 1`] = `
Object {
  "games": Object {
    "1": Object {
      "config": Object {
        "dndEnabled": false,
        "playersMaxCount": 4,
        "startingDelayInSec": 4,
      },
      "id": "1",
      "name": "createGame",
      "players": Object {
        "userOneId": Object {
          "cards": Array [],
          "id": "userOneId",
          "isHost": true,
          "ligrettoDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "stackDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "stackOpenDeck": Object {
            "cards": Array [],
            "isHidden": true,
          },
          "status": "DontReadyToPlay",
        },
      },
      "playground": Object {
        "decks": Array [],
        "droppedDecks": Array [],
      },
      "spectators": Object {},
      "status": "New",
    },
  },
  "users": Object {
    "userOneId": Object {
      "currentGameId": "1",
      "id": "userOneId",
      "socketIds": Array [
        "socket1",
        "socket2",
      ],
    },
  },
}
`;
